### PR TITLE
Replace IF..ELSE IF with SWITCH..CASE

### PR DESCRIPTION
**Pull request by khanrn**
I think `SWITCH..CASE..DEFAULT` is much simpler than `IF..ELSE IF..ELSE`. :slightly_smiling_face:

### CODE REVIEW

File: libevmasm/KnownState.cpp [hunk 1]
```diff
@@ -153,13 +159,17 @@ KnownState::StoreOperation KnownState::feedItem(AssemblyItem const& _item, bool
 					resetStorage();
 				if (invMem || invStor)
 					m_sequenceNumber += 2; // Increment by two because it can read and write
+
 				assertThrow(info.ret <= 1, InvalidDeposit, "");
 				if (info.ret == 1)
+				{
```
Comment (erak) [inline libevmasm/KnownState.cpp]: These are not needed here and can be removed again. In general, we're striving towards a coding style that supports displaying as many lines as possible on one screen.

File: libevmasm/KnownState.cpp [hunk 2]
```diff
@@ -153,13 +159,17 @@ KnownState::StoreOperation KnownState::feedItem(AssemblyItem const& _item, bool
...
 					setStackElement(
 						m_stackHeight + _item.deposit(),
 						m_expressionClasses->find(_item, arguments, _copyItem)
 					);
+				}
 			}
+
```
Comment (erak) [inline libevmasm/KnownState.cpp]: The above comment also applies to the new lines that you've added.

**Review by erak — CHANGES_REQUESTED** [review]
Please also change the commit message to something more meaningful.

**Review by chriseth — DISMISSED** [review]

**Comment by khanrn** [comment]
Some git issue happened when squashing. A cleaned #4398 PR has been submitted as a replacement of this PR.

**Pull request closed without merging**